tortoise/threshold.go

```diff
@@ -98,39 +98,20 @@ func computeThresholdForLayers(config Config, epochWeight map[types.EpochID]weig
 	return threshold
 }
 
-func getVerificationWindow(config Config, state *commonState, tmode mode) types.LayerID {
-	target := state.verified.Add(1)
-	window := state.last
-	if tmode.isFull() && state.last.Difference(target) > config.FullModeVerificationWindow {
-		window = target.Add(config.FullModeVerificationWindow)
-	} else if tmode.isVerifying() && state.last.Difference(target) > config.VerifyingModeVerificationWindow {
-		window = target.Add(config.VerifyingModeVerificationWindow)
+func getVerificationWindow(config Config, tmode mode, target, last types.LayerID) types.LayerID {
+	if tmode.isFull() && last.Difference(target) > config.FullModeVerificationWindow {
+		return target.Add(config.FullModeVerificationWindow)
+	} else if tmode.isVerifying() && last.Difference(target) > config.VerifyingModeVerificationWindow {
+		return target.Add(config.VerifyingModeVerificationWindow)
 	}
-	return window
+	return last
 }
 
-// updateThresholds recomputes local and global thresholds:
-// - when last layer is updated
-// - when verified layer is updated
-// - when switching from one mode into the other.
-func updateThresholds(logger log.Log, config Config, state *commonState, tmode mode) {
-	state.localThreshold = computeLocalThreshold(config, state.epochWeight, state.last)
+func computeThresholds(logger log.Log, config Config, tmode mode, target, last, processed types.LayerID, epochWeight map[types.EpochID]weight) (weight, weight) {
```

Comment:
Looking at the signature, it's not clear what thresholds are returned. What do you think of something like the below?
```suggestion
func computeThresholds(logger log.Log, config Config, tmode mode, target, last, processed types.LayerID, epochWeight map[types.EpochID]weight) (local, global weight) {
```

Reply:
makes sense :+1: